File: App backend/models/user.model.js
const mongoose = require("mongoose");
const db = require("../config/db");

const { Schema } = mongoose;

const userSchema = new Schema({
  username: {
    type: String,
    required: true,
  },
  email: {
    type: String,
    required: [true, "Email is required"],
    unique: true,
    lowercase: true,
    match: [/^\S+@\S+\.\S+$/, "Please enter a valid email"],
  },
  registration: {
    type: String,
    required: [true, "Registration number is required"],
    unique: true, 
  },
  password: {
    type: String,
    required: [true, "Password is required"],
    minlength: [6, "Password must be at least 6 characters long"],
  },
  role: {
    type: String,
    enum: ['user', 'admin','varsity'],
    default: 'user',
  },
  profilePicture: {
    type: String,// base64// Local default image path
    default: "null",
  },
  department:{
  type:String,
  default:"Software Engineering",
  },
  university:{
  type:String,
  default:"SUST",
  },
  coverPicture: {
      type: String,// base64
      default: "null", // Local default image path
    },
});

const UserModel = db.model('User', userSchema);

module.exports = UserModel;

File: Campus Clubs/lib/config/UserModel.dart
class UserModel{
  final String id;
  final String name;
  final String email;
  final String registation;
  final String profilePicture;
  final String department;
  final String university;
  final String role;
  final String coverPicture;


  UserModel({
    required this.id,
    required this.name,
    required this.email,
    required this.registation,
    required this.profilePicture,
    required this.role,
    required this.department,
    required this.university,
    required this.coverPicture,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      id: json['_id'] ?? '',
      name: json['username'] ?? '',  // backend uses username for name
      email: json['email'] ?? '',
      registation: json['registration'] ?? '',
      profilePicture: json['profilePicture'] ?? '',
      department: json['department'] ?? '',
      university: json['university'] ?? '',
      coverPicture : json['coverPicture']??'',
      role: json['role'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'username': name,
      'email': email,
      'registration': registation,
      'profilePicture': profilePicture,
      'coverPicture':coverPicture,
      'department': department,
      'university': university,
      'role': role,
    };
  }
}